### ISSUE TITLE
Allow resources without class_name to be created in the EditorResourcePicker

### ISSUE THREAD

**Issue by dugramen** [godotengine/godot-proposals#8102]
### Describe the project you are working on

Editor plugins

### Describe the problem or limitation you are having in your project

When exporting a resource, you get an EditorResourcePicker in the inspector. It shows a list of extended resources which can be created, but only ones that have an explicit class_name. Subsequently it seems like you can't have built-in anonymouse resources at all, since to create that resource you'd have to make a new file and drag that in

I'd like anonymouse resource classes so that it doesn't pollute the list of class names

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow anonymous resource classes to be created in the resource picker. This would enable anonymouse resources to be built in and not pollute the class list namespace or the filesystem

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I have 2 suggestions

1.  Include anonymous resource classes in the ResourcePicker list. Their 'name' would be the script's resource_path. Since that could be really long and ugly, make a submenu for anonymous resources under "more ->"

2.  Allow dragging and dropping a script into the resoruce picker, and if it inherits the proper resource, create a built in instance (a new resource with that script attatched). Of course, since scripts are also resources, ignore this feature when the resource picker expects a script

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Possibly? But it would be very hacky and not easy

### Is there a reason why this should be core and not an add-on in the asset library?

EditorResourcePicker is core

**Comment by AThousandShips** [comment]
What's the problem with naming the class?

**Comment by dugramen** [comment]
Naming the class isn't a huge problem or anything, but it does make things less sharable. I'll give the context of my project as an example

The plugin I'm working on is a custom stylebox that accepts a list of resources to change its properties via code. This is specifically to tackle the pain point of getting things like the 'accent_color' and 'base_color' from editor settings. Ideally someone would only have to share the theme.tres file to get the same theme, so it would be beneficial to have everything built in and anonymous, in case there is some naming clash

Again not a huge deal, but I think it should be possible at least

**Comment by KoBeWi** [comment]
You can create empty Resource and attach script.

**Comment by dugramen** [comment]
Ok so my main issue was that I thought built in resources was not possible without class_name, but I just realized that clicking "make unique" literally just makes a resource, even scripts, built in. So I can work with that

**Comment by Tattomoosa** [comment]
I think adding namespaces would be the ideal solution here as far as I'm concerned [(namespace proposal)](https://github.com/godotengine/godot-proposals/issues/1566) since `class_name` pollution/collision is the primary reason to avoid it, especially when developing addons.

I also would be for adding a "Load Script" option into the inspector's resource picker and the create resource dialog to facilitate creating new anonymous resources, but I think with namespaces there wouldn't be much incentive to have anonymous resources in the first place.

Something related to this though that I think is a bug is that resources with anonymous scripts (existing `.tres` files) cannot be loaded via Quick Load or Load, and must be drag and dropped from the FileSystem. Fixing that would go a long way towards making anonymous resources more usable.

**Comment by rvenson** [comment]
> You can create empty Resource and attach script.

It seems not possible to attach a resource to a node using the editor (drag-n-drop or EditorResourcePicker).

````gdscript
extends Resource
@export var name : String
@export var production_time : int
````

````gdscript
extends Node
@export var selected_recipe : Recipe
````

Via scripting, it works. There are any workarounds to this issue?

**Comment by KoBeWi** [comment]
Ii you export `Recipe`, the editor will accept only resources that have a script named Recipe attached, or its derivatives. For nameless scripts you'd need to export plain Resource.

**Comment by rvenson** [comment]
Yeah, I see. It misses the point of exporting the variable for me since it could lead to the wrong resource type being attached to the node or more code to type check. Class names still are not an option for me without namespaces. Thanks for the clarification.

**Comment by GreenCrowDev** [comment]
> I think adding namespaces would be the ideal solution here as far as I'm concerned [(namespace proposal)](https://github.com/godotengine/godot-proposals/issues/1566) since `class_name` pollution/collision is the primary reason to avoid it, especially when developing addons.

I agree that this issue is more about not having namespaces than anything else.

**Comment by Oblepikha** [comment]
For my resources I often have a base script with a `class_name` and many inherited scripts without names.
In the base class i expose a variable to attach any script:
```gdscript
@tool
class_name BaseResource extends Resource

func _get_property_list() -> Array[Dictionary]:
	var properties: Array[Dictionary]
	properties.append({
		"name": "my_script",
		"type": TYPE_OBJECT,
		"hint": PROPERTY_HINT_RESOURCE_TYPE,
		"hint_string": "Script",
		"usage": PROPERTY_USAGE_EDITOR
	})
	return properties

func _get(property: StringName) -> Variant:
	match property:
		"my_script": return get_script()
	return null

func _set(property: StringName, value: Variant) -> bool:
	match property:
		"my_script":
			set_script(value)
			return true
	return false
```
Would be nice to have a native option to attach a script to a resource - for example via drag and drop, an exposed script variable, or an "Attach Script" button similar to nodes
